### PR TITLE
feat(ci): collect and release binaries and image

### PR DESCRIPTION
**Pull request by derekbit**
#### Which issue(s) this PR fixes:

Issue Longhorn/longhorn#8980

#### What this PR does / why we need it:

Collect and release binaries and image

image

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -21,3 +24,90 @@ jobs:
...
         token: ${{ secrets.CODECOV_TOKEN }}
+
+    - name: Upload binaries
+      uses: actions/upload-artifact@v4
+      with:
+        name: binaries_artifact
+        path: ./bin/*
+
+   # Calculate sha256 and sha12 checksum
+    - name: Create sha256 checksum
```
Comment (innobead) [inline .github/workflows/build.yml:35]: One checksum for artifacts is good enough.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -21,3 +24,90 @@ jobs:
...
+      run: mkdir -p checksum && cd bin && for i in *; do sha256sum -b $i > "../checksum/$i.sha256"; done && cd -
+
+    - name: Create sha512 checksum
+      run: mkdir -p checksum && cd bin && for i in *; do sha512sum -b $i > "../checksum/$i.sha512"; done && cd -
+
+    # Upload binaries to release
+    - name: Upload binaries to release
```
Comment (innobead) [inline .github/workflows/build.yml:42]: This should be executed only for release/tag.

File: .github/workflows/build.yml (hunk 1 above)
Comment (derekbit) [reply .github/workflows/build.yml:35]: Addressed. Thank you.

File: .github/workflows/build.yml (hunk 2 above)
Comment (derekbit) [reply .github/workflows/build.yml:39]: Addressed. Thank you.

**Review by innobead — APPROVED** [review]

**Pull request merged**